test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/file-tailing/auto_multi_line_test.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package filetailing
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+)
+
+//go:embed log-config/automulti.yaml
+var agentAutoMultiLineConfig string
+
+const service = "hello"
+const singleLineLog = "This is a single line log"
+const multiLineLog = "This is\na multi\nline log"
+
+type AutoMultiLineSuite struct {
+	e2e.BaseSuite[environments.Host]
+	DevMode bool
+}
+
+func TestAutoMultiLineSuite(t *testing.T) {
+	s := &AutoMultiLineSuite{}
+	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awshost.Provisioner(awshost.WithAgentOptions(agentparams.WithLogs(), agentparams.WithAgentConfig(agentAutoMultiLineConfig), agentparams.WithIntegration("custom_logs.d", logConfig)))),
+	}
+	if devMode, err := strconv.ParseBool(devModeEnv); err == nil && devMode {
+		options = append(options, e2e.WithDevMode())
+	}
+
+	e2e.Run(t, s, options...)
+}
+
+func (s *AutoMultiLineSuite) TestAutoMultiLine() {
+	s.Run("AutoMultiLine", s.ContainsLogWithNewLines)
+}
+
+func (s *AutoMultiLineSuite) prePopulate() {
+	var message string
+
+	// auto_multi_line_detection uses the first 500 logs to detect a pattern
+	// Alternate between sending single line logs and multi line logs
+	for i := 0; i < 700; i++ {
+		timestamp := time.Now().Format(time.RFC3339)
+		if i%2 == 0 {
+			message = fmt.Sprintf("%s | %s", timestamp, singleLineLog)
+		} else {
+			message = fmt.Sprintf("%s | %s", timestamp, multiLineLog)
+		}
+		cmd := fmt.Sprintf("echo '%s' | sudo tee -a %s", message, logPath)
+		s.Env().RemoteHost.MustExecute(cmd)
+	}
+}
+
+func (s *AutoMultiLineSuite) BeforeTest(suiteName, testName string) {
+	s.BaseSuite.BeforeTest(suiteName, testName)
+
+	s.Env().RemoteHost.Execute("sudo touch /var/log/hello-world.log")
+	s.Env().RemoteHost.Execute("sudo chmod +r /var/log/hello-world.log")
+
+	// Generate some logs for auto_multi_line_detection to kick in
+	s.prePopulate()
```

Comment:
NIT/suggestion: if we rename this function to something like `generateMultilineLogs` I think you can remove the above comment since it becomes self explanatory :)